reventlov/bot_plugins.py

```diff
@@ -0,0 +1,99 @@
+from future.utils import iteritems
+import os
+import logging
+import importlib
+import pkgutil
+
+logger = logging.getLogger(__name__)
+
+
+def iter_namespaces():
+    return pkgutil.iter_modules(
+        [os.path.join(os.path.dirname(__file__), 'plugins')],
+        'reventlov.plugins.',
+    )
+
+
+def import_plugin_modules():
+    modules = {
+        name: importlib.import_module(name)
+        for finder, name, ispkg
+        in iter_namespaces()
+    }
+    return modules
+
+
+class BotPlugins(object):
+    def __init__(self, dispatcher):
+        self.dispatcher = dispatcher
+        self.modules = import_plugin_modules()
+        self.plugins = {}
+        self.disabled_plugins = []
```

Comment:
What is the purpose of this attribute if you are using another private one `__disabled_plugins`?

Reply:
@edubxb comment was about having a public setter for `disabled_plugins` to be used only in the `__init__` of the owner class, so I replaced this with a private method.